tools/codegen/api/structured.py

```diff
@@ -102,10 +102,14 @@ def impl_arguments(g: NativeFunctionsGroup) -> List[Binding]:
                 non_out_args_replaced.append(a)
 
         args.extend(non_out_args_replaced)
+        # precomputed.add is the list of parameters that are added
+        # without replacement
+        precomputed_add_args = g.out.precomputed.add
```

Comment:
nit: this is only relevant if `g.out.precomputed` is set, but it's defined and used outside of this block.  Maybe you can just fold this into args directly here?
```
args.extend(g.out.precomputed.add)
args.extend(non_out_args_replaced)
```

That would also mean that the "added" arguments show up right before the other precomputed arguments in the schema, which would be consistent with the order of the precompute args from the yaml (since the optional first line is the added args)

Reply:
Yes, the problem of this is that the added arguments show up before any another arguments in the schema (even before the self tensor) because `non_out_args_replaced` actually contains all the arguments of the schema (the ones that are precomputed and the others that are not precomputed).

Reply:
Ah right.

What do you think of the order:
- precompute replaced args
- precompute added args
- out args

That way we still get `input` tensors at the beginning and `output` tensors at the end, which is the case for all other structured kernels today.

Either way, if we're putting `precompute.add` args after the replace args, then maybe we should make the "add" arguments in native_functions.yaml the (optional) *last* line instead of the first :)

Reply:
Perfect to me :)

I will fix everything today!